src/js/pages/Setup.js

```diff
@@ -533,31 +535,19 @@ class Setup extends mixin(StoreMixin) {
 
   getValidationFn(key, type) {
     return (fieldValue) => {
-      if ((key === 'master_list' || key === 'agent_list') && (fieldValue != null || fieldValue !== '')) {
-        this.clearLocalValidationItem(key);
-      }
-
-      if (type === 'port' && fieldValue != null && fieldValue !== '') {
-        if (parseInt(fieldValue) > 65535) {
-          let localValidationErrors = this.state.localValidationErrors;
-          localValidationErrors[key] = 'Ports must be less than or equal to 65535.';
-          this.setState({localValidationErrors});
-
-          return false;
-        }
+      let isValueEmpty = fieldValue == null || fieldValue === '';
+      let isValidatedLocally = key === 'master_list' || key === 'agent_list'
+        || key === 'public_ip_address' || type === 'port';
 
+      if (isValidatedLocally && !isValueEmpty) {
         this.clearLocalValidationItem(key);
-      } else if (key === 'public_ip_address' && fieldValue != null && fieldValue !== '') {
-        if (PUBLIC_HOSTNAME_VALIDATION.test(fieldValue) || fieldValue === '' || fieldValue == null) {
-          return true;
-        }
-
-        let localValidationErrors = this.state.localValidationErrors;
-        localValidationErrors[key] = 'Enter a valid IP address.';
-        this.setState({localValidationErrors});
+      }
 
-        return false;
-      } else if (key === 'public_ip_address' && (fieldValue == null || fieldValue === '')) {
+      if (type === 'port' && !isValueEmpty) {
+        return this.isPortValid(fieldValue, key);
+      } else if (key === 'public_ip_address' && !isValueEmpty) {
```

Comment:
These `else`'s are not useful since we return in each of these statements. Let's just use `if`'s; easier to reason about and read